feat(combo-box): introduce combo box

asudoh opened the pull request:
Fixes #44.

cal-smith commented on src/components/dropdown/dropdown.ts:
```diff
@@ -506,15 +548,14 @@ class BXDropdown extends HostListenerMixin(FocusMixin(LitElement)) {
         <div
           role="button"
```
Just want to verify that `role="button"` is valid if the trigger isn't focussable ...

asudoh replied:
Good point! Fixed.

asudoh commented:
Thank you for reviewing and finding the issues @cal-smith!

> - If I'm typing and press space it auto selects the first item

Couldn't reproduce it with a space key. Reproduced it with hitting backspace key to make the `<input>` empty and I guess this is what you meant, but please don't hesitate to correct me if I'm wrong here.

> - when I arrow key up and down the cursor moves to the start and end of the text, as well as moving the selection ... I'm not sure what's correct here, but that feels wrong

Makes sense, fixed.

> - when I clear the filter (with backspace) it doesn't always clear the selection right away (make a selection -> close the menu -> open the menu -> clear selection -> `x` still visible until tab out)

Great catch! Fixed.

cal-smith commented:
> Couldn't reproduce it with a space key. Reproduced it with hitting backspace key to make the <input> empty and I guess this is what you meant, but please don't hesitate to correct me if I'm wrong here.

I'm still seeing it ... the easiest way to replicate seems to be to start typing an option title, then press space, like so:

![giphy](https://user-images.githubusercontent.com/1744185/62723803-8a42f400-b9df-11e9-81c2-2bec519f2110.gif)

asudoh commented:
Got it - Thanks alot @cal-smith for clarifying! Fixed.
